### PR TITLE
FIX: Use delete_all_posts_max to improve delete user protection consistency 

### PR DESCRIPTION
**Pull request by romanrizzi**
cc @techAPJ

### CODE REVIEW

File: app/assets/javascripts/admin/controllers/admin-user-index.js.es6 [hunk 1]
```diff
@@ -199,7 +199,8 @@ export default Controller.extend(CanCheckEmails, {
 
     destroy() {
       const postCount = this.get("model.post_count");
-      if (postCount <= 5) {
+      const maxPostCount = Discourse.SiteSettings.delete_all_posts_max;
```
Comment (eviltrout) [inline app/assets/javascripts/admin/controllers/admin-user-index.js.es6]: Can this be `this.siteSettings.delete_all_posts_max` instead?

File: app/assets/javascripts/discourse/models/user.js [hunk 2]
```diff
@@ -45,7 +45,8 @@ const User = RestModel.extend({
 
   @discourseComputed("can_be_deleted", "post_count")
   canBeDeleted(canBeDeleted, postCount) {
-    return canBeDeleted && postCount <= 5;
+    const maxPostCount = Discourse.SiteSettings.delete_all_posts_max;
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/models/user.js]: ditto here

Comment (romanrizzi) [reply app/assets/javascripts/discourse/models/user.js]: `this.siteSettings` is undefined here 🤔

Comment (eviltrout) [reply app/assets/javascripts/discourse/models/user.js]: Ah yes it's a model. Can't avoid this here for now unless you set it when the model is built. Fine to ignore this one.

**Pull request merged**